src/pack.c

```diff
@@ -490,6 +490,12 @@ int git_packfile_resolve_header(
 
 		base_offset = get_delta_base(p, &w_curs, &curpos, type, offset);
 		git_mwindow_close(&w_curs);
+
+		if (base_offset < 0) {
+			error = (int)base_offset;
+			return error;
+		}
```

Comment:
These are a bit awkward and don't really fit into our greater error handling patterns. Would you mind updating `get_delta_base` to return the base offset via an out parameter and have the direct return value indicate errors, only? Let me know if that exceeds the time you want to invest here

Reply:
I would not mind! I also found these a bit awkward and I was aiming for the change with the smallest possible delta, so I thought that a refactor in the same change was not ideal.

But if that's a possibility, I would also prefer to do the cleanup now.

Done!